src/bsd.rs

```diff
@@ -0,0 +1,46 @@
+use std::fs::read_link;
+use std::path::PathBuf;
+use libc::*;
+use std::ptr;
+
+// http://stackoverflow.com/q/799679
+// http://stackoverflow.com/a/1024937
+pub fn get_executable_path() -> Option<PathBuf> {
+    // FreeBSD
+    let mut mib = [0; 4];
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC;
+    mib[2] = KERN_PROC_PATHNAME;
+    mib[3] = -1;
+    let buf = [0; 1024];
+    let mut cb = 1024 as size_t;
+    let result = unsafe {
+        sysctl(mib.as_ptr(),
+               4,
+               buf.as_ptr() as *mut c_void,
+               &mut cb as *mut size_t,
+               ptr::null(),
+               0)
+    };
+
+    if result == 0 {
+        // Convert the string allocated on the stack to a Rust string.
+        let mut path = String::new();
+        let len = unsafe { strlen(buf.as_ptr()) };
+        let mut i = 0;
+        while i < len {
+            path.push(buf[i] as u8 as char);
+            i = i + 1;
+        }
+        let mut pb = PathBuf::new();
+        pb.push(path);
+        Some(pb)
+    } else {
+        // NetBSD
```

Comment:
This says NetBSD but the module declaration in `lib.rs` says OpenBSD. Is that correct?